NullPointerException on BasicMapzenActivity.onDestroy(

While trying to test #468 ran into the following exception when trying to aggressively rotate between landscape and portrait mode:


```
FATAL EXCEPTION: main
                                                                               Process: com.mapzen.android.sdk.sample, PID: 21126
                                                                               java.lang.RuntimeException: Unable to destroy activity {com.mapzen.android.sdk.sample/com.mapzen.android.sdk.sample.BasicMapzenActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.mapzen.android.graphics.MapzenMap.setFindMeOnClickListener(android.view.View$OnClickListener)' on a null object reference
                                                                                   at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4639)
                                                                                   at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4662)
                                                                                   at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4950)
                                                                                   at android.app.ActivityThread.-wrap21(ActivityThread.java)
                                                                                   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1656)
                                                                                   at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                                   at android.os.Looper.loop(Looper.java:154)
                                                                                   at android.app.ActivityThread.main(ActivityThread.java:6688)
                                                                                   at java.lang.reflect.Method.invoke(Native Method)
                                                                                   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1468)
                                                                                   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1358)
                                                                                Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.mapzen.android.graphics.MapzenMap.setFindMeOnClickListener(android.view.View$OnClickListener)' on a null object reference
                                                                                   at com.mapzen.android.sdk.sample.BasicMapzenActivity.onDestroy(BasicMapzenActivity.java:92)
                                                                                   at android.app.Activity.performDestroy(Activity.java:7165)
                                                                                   at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1161)
                                                                                   at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4626)
                                                                                   at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4662) 
                                                                                   at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4950) 
                                                                                   at android.app.ActivityThread.-wrap21(ActivityThread.java) 
                                                                                   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1656) 
                                                                                   at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                                   at android.os.Looper.loop(Looper.java:154) 
                                                                                   at android.app.ActivityThread.main(ActivityThread.java:6688) 
                                                                                   at java.lang.reflect.Method.invoke(Native Method) 
                                                                                   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1468) 
                                                                                   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1358) 
```

--- a/samples/mapzen-android-sdk-sample/src/main/java/com/mapzen/android/sdk/sample/BasicMapzenActivity.java
+++ b/samples/mapzen-android-sdk-sample/src/main/java/com/mapzen/android/sdk/sample/BasicMapzenActivity.java
@@ -89,7 +89,9 @@ public class BasicMapzenActivity extends BaseDemoActivity
 
   @Override protected void onDestroy() {
     super.onDestroy();
-    map.setFindMeOnClickListener(null);
+    if (map != null) {
+      map.setFindMeOnClickListener(null);
+    }
   }
 
   @Override public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {

--- a/samples/mapzen-android-sdk-sample/src/main/java/com/mapzen/android/sdk/sample/MapzenSearchViewActivity.java
+++ b/samples/mapzen-android-sdk-sample/src/main/java/com/mapzen/android/sdk/sample/MapzenSearchViewActivity.java
@@ -1,4 +1,4 @@
-package com.mapzen.android.sdk.sample;
+ package com.mapzen.android.sdk.sample;
 
 import com.mapzen.android.graphics.MapView;
 import com.mapzen.android.graphics.MapzenMap;
@@ -99,6 +99,8 @@ public class MapzenSearchViewActivity extends BaseDemoActivity {
 
   @Override protected void onDestroy() {
     super.onDestroy();
-    mapzenMap.setPersistMapData(false);
+    if (mapzenMap != null) {
+        mapzenMap.setPersistMapData(false);
+    }
   }
 }

--- a/samples/mapzen-android-sdk-sample/src/main/java/com/mapzen/android/sdk/sample/MarkerMapzenActivity.java
+++ b/samples/mapzen-android-sdk-sample/src/main/java/com/mapzen/android/sdk/sample/MarkerMapzenActivity.java
@@ -40,7 +40,9 @@ public class MarkerMapzenActivity extends BaseDemoActivity {
   }
 
   @Override protected void onDestroy() {
-    map.removeMarker();
+    if (map != null) {
+        map.removeMarker();
+    }
     super.onDestroy();
   }
 }

--- a/samples/mapzen-android-sdk-sample/src/main/java/com/mapzen/android/sdk/sample/OverlayActivity.java
+++ b/samples/mapzen-android-sdk-sample/src/main/java/com/mapzen/android/sdk/sample/OverlayActivity.java
@@ -57,7 +57,9 @@ public class OverlayActivity extends SwitchStyleActivity {
   }
 
   @Override protected void onDestroy() {
-    mapzenMap.setMyLocationEnabled(false);
+    if (mapzenMap != null) {
+        mapzenMap.setMyLocationEnabled(false);
+    }
     super.onDestroy();
   }
 

--- a/samples/mapzen-android-sdk-sample/src/main/java/com/mapzen/android/sdk/sample/PolygonMapzenActivity.java
+++ b/samples/mapzen-android-sdk-sample/src/main/java/com/mapzen/android/sdk/sample/PolygonMapzenActivity.java
@@ -43,7 +43,9 @@ public class PolygonMapzenActivity extends BaseDemoActivity {
   }
 
   @Override protected void onDestroy() {
-    map.removePolygon();
+    if (map != null) {
+        map.removePolygon();
+    }
     super.onDestroy();
   }
 }

--- a/samples/mapzen-android-sdk-sample/src/main/java/com/mapzen/android/sdk/sample/PolylineMapzenActivity.java
+++ b/samples/mapzen-android-sdk-sample/src/main/java/com/mapzen/android/sdk/sample/PolylineMapzenActivity.java
@@ -42,7 +42,9 @@ public class PolylineMapzenActivity extends BaseDemoActivity {
   }
 
   @Override protected void onDestroy() {
-    map.removePolyline();
+    if (map != null) {
+        map.removePolyline();
+    }
     super.onDestroy();
   }
 }

--- a/samples/mapzen-android-sdk-sample/src/main/java/com/mapzen/android/sdk/sample/RoutePinsActivity.java
+++ b/samples/mapzen-android-sdk-sample/src/main/java/com/mapzen/android/sdk/sample/RoutePinsActivity.java
@@ -61,7 +61,9 @@ public class RoutePinsActivity extends BaseDemoActivity {
   }
 
   @Override protected void onDestroy() {
-    map.clearRoutePins();
+    if (map != null) {
+        map.clearRoutePins();
+    }
     super.onDestroy();
   }
 }

--- a/samples/mapzen-android-sdk-sample/src/main/java/com/mapzen/android/sdk/sample/RouterActivity.java
+++ b/samples/mapzen-android-sdk-sample/src/main/java/com/mapzen/android/sdk/sample/RouterActivity.java
@@ -118,7 +118,9 @@ public class RouterActivity extends BaseDemoActivity {
   }
 
   @Override protected void onDestroy() {
-    map.setMyLocationEnabled(false);
+    if (map != null) {
+        map.setMyLocationEnabled(false);
+    }
     super.onDestroy();
   }
 }

--- a/samples/mapzen-android-sdk-sample/src/main/java/com/mapzen/android/sdk/sample/SwitchStyleActivity.java
+++ b/samples/mapzen-android-sdk-sample/src/main/java/com/mapzen/android/sdk/sample/SwitchStyleActivity.java
@@ -86,7 +86,9 @@ public class SwitchStyleActivity extends BaseDemoActivity {
   @Override protected void onDestroy() {
     styleSpinner.setOnItemSelectedListener(null);
     updateBtn.setOnClickListener(null);
-    mapzenMap.setMyLocationEnabled(false);
+    if (mapzenMap != null) {
+        mapzenMap.setMyLocationEnabled(false);
+    }
     super.onDestroy();
   }
 

